packages/models/src/plugins/client/client-hook.model.ts

```diff
@@ -193,3 +210,17 @@ export type ClientHookName = keyof typeof clientHookObject
 export interface ClientHook {
   runHook <T> (hookName: ClientHookName, result?: T, params?: any): Promise<T>
 }
+
+export const clientDoActionObject = {
+  'admin-abuse-list:load-data': true,
+  'application:increment-loader': true,
+  'application:decrement-loader': true,
+  'admin-user-list:load-data': true,
+  'admin-video-comment-list:load-data': true,
+  'video-watch-comment-list:load-data': true
+}
+
+export type ClientDoActionName = keyof typeof clientDoActionObject
+
+export type ClientDoActionCallback = () => Observable<any>
```

Comment:
Replace `Observable` by a `Promise` here. We don't want to force to use `Observable`, and we can easily convert a `Promise` to an `Observable` in the client